snuba/api/query.py

```diff
@@ -89,84 +78,74 @@ def raw_query(
-                                sql,
-                                query_settings,
-                                # All queries should already be deduplicated at this point
-                                # But the query_id will let us know if they aren't
-                                query_id=query_id if use_deduper else None,
-                                with_totals=request.query.has_totals(),
-                            )
-                            status = 200
-
-                            logger.debug(sql)
-                            timer.mark('execute')
-                            stats.update({
-                                'result_rows': len(result['data']),
-                                'result_cols': len(result['meta']),
-                            })
-
-                            if use_cache:
-                                state.set_result(query_id, result)
-                                timer.mark('cache_set')
-
-                        except BaseException as ex:
-                            error = str(ex)
-                            status = 500
-                            logger.exception("Error running query: %s\n%s", sql, error)
-                            if isinstance(ex, ClickHouseError):
-                                result = {'error': {
-                                    'type': 'clickhouse',
-                                    'code': ex.code,
-                                    'message': error,
-                                }}
-                            else:
-                                result = {'error': {
-                                    'type': 'unknown',
-                                    'message': error,
-                                }}
-
-                    else:
-                        status = 429
-                        Reason = namedtuple('reason', 'scope name val limit')
-                        reasons = [
-                            Reason('global', 'concurrent', g_concurr, gcl),
-                            Reason('global', 'per-second', g_rate, grl),
-                            Reason('project', 'concurrent', p_concurr, pcl),
-                            Reason('project', 'per-second', p_rate, prl)
-                        ]
-                        reason = next((r for r in reasons if r.limit is not None and r.val > r.limit), None)
-                        result = {'error': {
-                            'type': 'ratelimit',
-                            'message': 'rate limit exceeded',
-                            'detail': reason and '{r.scope} {r.name} of {r.val:.0f} exceeds limit of {r.limit:.0f}'.format(r=reason)
-                        }}
+                    project_rate_limit_stats = rate_limit_stats_container.get_stats(PROJECT_RATE_LIMIT_NAME)
+
+                    if 'max_threads' in query_settings and \
+                            project_rate_limit_stats is not None \
```

Comment:
nit: could you please put "and" either at the beginning or at the end of the line consistently in both lines ?